Source/CesiumRuntime/Private/CesiumMetadataPickingBlueprintLibrary.cpp

```diff
@@ -148,10 +150,69 @@ bool UCesiumMetadataPickingBlueprintLibrary::FindUVFromHit(
   return true;
 }
 
+namespace {
+std::optional<TMap<FString, FCesiumMetadataValue>>
+getInstancePropertyTableValues(const FHitResult& Hit, int64 FeatureIDSetIndex) {
+  const auto* pInstancedComponent =
+      Cast<UCesiumGltfInstancedComponent>(Hit.Component);
+  if (!IsValid(pInstancedComponent)) {
+    return std::nullopt;
+  }
+  const TSharedPtr<FCesiumInstanceFeatures> pInstanceFeatures =
+      pInstancedComponent->pInstanceFeatures;
+  if (!pInstanceFeatures) {
+    return TMap<FString, FCesiumMetadataValue>();
+  }
```

Comment:
See @kring 's comment of course, but if we're keeping these lines then these two `if` statements need to return the same thing. The point is that getting the instance metadata values fails because ultimately the component doesn't contain (proper) instance features. So the behavior should be consistent either way.

Reply:
This was not clear and at least deserved a comment. The intent was to return std::nullopt if the primitive was not an instanced static mesh and therefore cause GetPropertyTableValuesFromHit() to use feature IDs from a primitive attribute. I was thinking that if the primitive was instanced and didn't have instance feature IDs, then it couldn't have feature IDs at all. But given our discussion above, that's not clearly the case.

Reply:
I added a comment about the different return values and I hope it's clear now. The return of an empty `TMap` indicates a problem with the instance metadata; in that case I think it's appropriate to not continue searching for other metadata.